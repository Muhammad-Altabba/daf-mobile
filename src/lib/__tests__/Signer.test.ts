import {
  resolvers,
  getDidsQuery,
  importSeedMutation,
  deleteSeedMutation,
  createDidMutation,
  DEFAULT_LEVEL,
  SHOW_SEED_PROMPT,
} from '../Signer'

import { RNUportHDSigner } from 'react-native-uport-signer'
jest.mock('../GraphQL')
jest.mock('../Log')

it('dids query returns valid result', async () => {
  const result = await resolvers.Query.dids(null, null, null)
  expect(RNUportHDSigner.listSeedAddresses).toBeCalled()
  expect(result).toEqual([
    { did: 'did:ethr:0x12345', address: '0x12345', __typename: 'Did' },
  ])
})

it('createSeed mutation returns new did', async () => {
  const result = await resolvers.Mutation.createDid(null, null, null)
  expect(RNUportHDSigner.createSeed).toBeCalledWith(DEFAULT_LEVEL)
  expect(result).toEqual({
    did: 'did:ethr:0x12345',
    address: '0x12345',
    __typename: 'Did',
  })
})

it('importSeed mutation returns new did', async () => {
  const args = { seed: 'test' }
  const result = await resolvers.Mutation.importSeed(null, args, null)
  expect(RNUportHDSigner.importSeed).toBeCalledWith(args.seed, DEFAULT_LEVEL)
  expect(result).toEqual({
    did: 'did:ethr:0x12345',
    address: '0x12345',
    __typename: 'Did',
  })
})

it('deleteSeed mutation returns bool', async () => {
  const args = { address: '0x12345' }
  const result = await resolvers.Mutation.deleteSeed(null, args, null)
  expect(RNUportHDSigner.deleteSeed).toBeCalledWith(args.address)
  expect(result).toEqual(true)
})

it('Did.seed query', async () => {
  const args = { address: '0x12345' }
  const result = await resolvers.Did.seed(args, null, null)
  expect(RNUportHDSigner.showSeed).toBeCalledWith(
    args.address,
    SHOW_SEED_PROMPT,
  )
  expect(result).toEqual('test')
})

it('query snapshots', async () => {
  expect(getDidsQuery).toMatchSnapshot()
  expect(importSeedMutation).toMatchSnapshot()
  expect(deleteSeedMutation).toMatchSnapshot()
  expect(createDidMutation).toMatchSnapshot()
})
