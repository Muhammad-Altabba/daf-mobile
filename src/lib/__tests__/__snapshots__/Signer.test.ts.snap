// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query snapshots 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "getDids",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [
              Object {
                "arguments": Array [],
                "kind": "Directive",
                "name": Object {
                  "kind": "Name",
                  "value": "client",
                },
              },
            ],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "dids",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "did",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "address",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "seed",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "isSelected",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
          },
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [
              Object {
                "arguments": Array [],
                "kind": "Directive",
                "name": Object {
                  "kind": "Name",
                  "value": "client",
                },
              },
            ],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "selectedDid",
            },
            "selectionSet": undefined,
          },
        ],
      },
      "variableDefinitions": Array [],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 124,
    "start": 0,
  },
}
`;

exports[`query snapshots 2`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "importSeed",
      },
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "seed",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "seed",
                  },
                },
              },
            ],
            "directives": Array [
              Object {
                "arguments": Array [],
                "kind": "Directive",
                "name": Object {
                  "kind": "Name",
                  "value": "client",
                },
              },
            ],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "importSeed",
            },
            "selectionSet": undefined,
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "seed",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 81,
    "start": 0,
  },
}
`;

exports[`query snapshots 3`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "deleteSeed",
      },
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "address",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "address",
                  },
                },
              },
            ],
            "directives": Array [
              Object {
                "arguments": Array [],
                "kind": "Directive",
                "name": Object {
                  "kind": "Name",
                  "value": "client",
                },
              },
            ],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "deleteSeed",
            },
            "selectionSet": undefined,
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "address",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 90,
    "start": 0,
  },
}
`;

exports[`query snapshots 4`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "createDid",
      },
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [
              Object {
                "arguments": Array [],
                "kind": "Directive",
                "name": Object {
                  "kind": "Name",
                  "value": "client",
                },
              },
            ],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "createDid",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "did",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 68,
    "start": 0,
  },
}
`;
